frank/fit.py

```diff
@@ -0,0 +1,275 @@
+    """
+
+    import argparse
+
+    parser = argparse.ArgumentParser("Run a Frank fit, by default using"
+                                     " parameters in default_parameters.json")
+    parser.add_argument("-p", "--parameter_filename",
+                        default='default_parameters.json', type=str,
+                        help="Parameter file (.json; see frank.fit.helper)")
+    parser.add_argument("-uv", "--uvtable_filename", default=None, type=str,
+                        help="UVTable file with data to be fit. See"
+                             " frank.io.load_uvtable")
+
+    args = parser.parse_args()
+    model = json.load(open(args.parameter_filename, 'r'))
+
+    if args.uvtable_filename:
+        model['input_output']['uvtable_filename'] = args.uvtable_filename
+
+    if ('uvtable_filename' not in model['input_output'] or
+        not model['input_output']['uvtable_filename']):
+        raise ValueError("    uvtable_filename isn't specified."
+                 " Set it in the parameter file or run frank with"
+                 " python -m frank.fit -uv <uvtable_filename>")
+
+    if not model['input_output']['load_dir']:
+        model['input_output']['load_dir'] = os.getcwd()
+
+    if not model['input_output']['save_dir']:
+        model['input_output']['save_dir'] = model['input_output']['load_dir']
+
+    logging.basicConfig(level=logging.INFO,
+        format='%(message)s',
+        handlers=[
+        logging.FileHandler(model['input_output']['save_dir'] +
+        '/frank_fit.log', mode='w'), logging.StreamHandler()]
+        )
+
+    logging.info('\nRunning frank on %s'
+                 %model['input_output']['uvtable_filename'])
+
+    logging.info('  Saving parameters to be used in fit to'
+                 ' `frank_used_pars.json`')
+    with open('frank_used_pars.json', 'w') as f:
+        json.dump(model, f, indent=4)
+
+    return model
+
+
+def load_data(data_file):
+    """
+    Read in a UVTable with data to be fit. See frank.io.load_uvtable
+    """
+
+    logging.info('  Loading UVTable')
+
+    u, v, vis, weights = frank.io.load_uvtable(data_file)
+
+    return u, v, vis, weights
+
+
+def determine_geometry(model, u, v, vis, weights):
+    """
+    Determine the source geometry (inclination, position angle, phase offset).
+
+    Parameters
+    ----------
+    model : dict
+          Dictionary containing model parameters the fit uses
+    u, v : array, unit = :math:`\\lambda`
+          u and v coordinates of observations
+    vis : array, unit = Jy
+          Real component of observed visibilities
+    weights : array, unit = Jy^-2
+          Weights assigned to observed visibilities, of the form
+          :math:`1 / \\sigma^2`
+
+    Returns
+    -------
+    geom : SourceGeometry object
+          Fitted geometry (see frank.geometry.SourceGeometry)
+    """
+
+    logging.info('  Determining disc geometry')
+
+    if not model['geometry']['fit_geometry']:
```

Comment:
We should explicitly require the user to provide 0,0,0,0 in the parameter file rather than handling this as a special case. There are a few reasons for this

1.  We don't have to maintain two bits of code
2. We avoid possible confusion
3. in the library we opted to not have a 'default geometry', we shouldn't have one here either

I would also prefer a syntax like:
```
model['geometry']['type'] = { 'known', 'fit_gaussian', } 
```
This is more readily extendible without needing to alter existing interfaces and also avoids the possibility of multiple ways of fitting the geometry specified simultaneously.